.github/workflows/application.yml

```diff
@@ -0,0 +1,343 @@
+
+jobs:
+    static-checks:
+        runs-on: ubuntu-latest
+        
+        name: "Static checks (PHP ${{ matrix.php }}, Symfony ${{ matrix.symfony }})"
+
+        strategy:
+            fail-fast: false
+            matrix:
+                php: [7.3, 7.4]
+                symfony: [4.4.*]
+                
+        steps:
+            -   
+                uses: actions/checkout@v2
+
+            -   
+                name: Setup PHP
+                uses: shivammathur/setup-php@v2
+                with:
+                    php-version: "${{ matrix.php }}"
+                    ini-values: date.timezone=Europe/Warsaw, opcache.enable=1, opcache.enable_cli=1, opcache.memory_consumption=256, opcache.max_accelerated_files=32531, opcache.interned_strings_buffer=8, opcache.validate_timestamps=0, opcache.save_comments=1, opcache.fast_shutdown=0
+                    extensions: intl, gd, opcache, mysql, pdo_mysql, :xdebug
+                    tools: symfony
+                    coverage: none
+
+            -   
+                name: Restrict Symfony version
+                if: matrix.symfony != ''
+                run: composer config extra.symfony.require "${{ matrix.symfony }}"
+
+            -   
+                name: Get Composer cache directory
+                id: composer-cache
+                run: echo "::set-output name=dir::$(composer config cache-files-dir)"
+
+            -   
+                name: Cache Composer
+                uses: actions/cache@v2
+                with:
+                    path: ${{ steps.composer-cache.outputs.dir }}
+                    key: ${{ runner.os }}-php-${{ matrix.php }}-symfony-${{ matrix.symfony }}-composer-${{ hashFiles('**/composer.json **/composer.lock') }}
+                    restore-keys: |
+                        ${{ runner.os }}-php-${{ matrix.php }}-symfony-${{ matrix.symfony }}-composer-
+
+            -   
+                name: Install PHP dependencies
+                run: composer update --no-interaction --no-scripts
+
+            -   
+                name: Validate composer.json
+                run: composer validate --strict
+                
+            -
+                name: Check for security vulnerabilities
+                run: symfony security:check
+                
+            -
+                name: Validate Doctrine mapping
+                run: bin/console doctrine:schema:validate --skip-sync -vvv
+                
+            -
+                name: Validate Twig templates
+                run: bin/console lint:twig src
+                
+            -
+                name: Validate Yaml files
+                run: bin/console lint:yaml src
+                
+            -
+                name: Run Psalm
+                run: vendor/bin/psalm --show-info=false --output-format=github
+                
+            -
+                name: Run PHPStan
+                run: vendor/bin/phpstan analyse -c phpstan.neon -l 1 src
```

Comment:
As you run several independent test commands in the same job, there is a trick to make them run even if the others are failing (while still making the job fail so not using `continue-on-error`), so that you get the result from all of them in a single execution:

```diff
             -
                 name: Install PHP dependencies
                 run: composer update --no-interaction --no-scripts
+                id: end-of-setup

             -
                 name: Check for security vulnerabilities
                 run: symfony security:check
+                if: always() && steps.end-of-setup.outcome == 'success'

             -
                 name: Validate Doctrine mapping
                 run: bin/console doctrine:schema:validate --skip-sync -vvv
+                if: always() && steps.end-of-setup.outcome == 'success'
```

the `always()` part of the condition is necessary because any `if` expression which does not include one of the status check function receives an implicit `success()` check. The other part is about checking that the setup was successful

Reply:
Wow, that's cool! I'll open a PR soon.

Reply:
that was exactly my reaction when I discovered it :smile: